include/util/packed_vector.hpp

```diff
@@ -89,6 +92,17 @@ inline WordT set_upper_value(WordT word, WordT mask, std::uint8_t offset, T valu
     return (word & ~mask) | ((static_cast<WordT>(value) >> offset) & mask);
 }
 
+inline bool compare_and_swap(uint64_t *ptr, uint64_t old_value, uint64_t new_value)
```

Comment:
Tbh I am not sure if we need PackedVector to be thread-safe at all - after brief look I couldn't find a place where we modify it concurrently.

Reply:
According to #4199 it can happen during edge updates, probably when running `osrm-customize`

Reply:
As I can see set_value is used on push_back to - curious if we are safe on reallocations there 🤔